MdeModulePkg/Core/Dxe/Misc/MemoryProtection.c

```diff
@@ -1120,37 +1131,38 @@ CoreInitializeMemoryProtection (
   // - EfiConventionalMemory and EfiBootServicesData should use the
   //   same attribute
   //
-  // MU_CHANGE START: We allow code types to have NX
+  // MU_CHANGE START: We allow code types to have NX and EfiBootServicesData to differ in attributes from
+  //                  EfiConventionalMemory
   // ASSERT ((GetPermissionAttributeForMemoryType (EfiBootServicesCode) & EFI_MEMORY_XP) == 0);
   // ASSERT ((GetPermissionAttributeForMemoryType (EfiRuntimeServicesCode) & EFI_MEMORY_XP) == 0);
   // ASSERT ((GetPermissionAttributeForMemoryType (EfiLoaderCode) & EFI_MEMORY_XP) == 0);
+  // ASSERT (
+  //   GetPermissionAttributeForMemoryType (EfiBootServicesData) ==
+  //   GetPermissionAttributeForMemoryType (EfiConventionalMemory)
+  //   );
   // MU_CHANGE END
-  ASSERT (
-    GetPermissionAttributeForMemoryType (EfiBootServicesData) ==
-    GetPermissionAttributeForMemoryType (EfiConventionalMemory)
-    );
 
-  Status = CoreCreateEvent (
-             EVT_NOTIFY_SIGNAL,
-             // MU_CHANGE START: Use Project Mu Arch Protocol Notify
-             TPL_CALLBACK - 1,
-             //  MemoryProtectionCpuArchProtocolNotify,
-             MemoryProtectionCpuArchProtocolNotifyMu,
-             // MU_CHANGE END
-             NULL,
-             &Event
-             );
-  ASSERT_EFI_ERROR (Status);
-
-  //
-  // Register for protocol notifactions on this event
-  //
-  Status = CoreRegisterProtocolNotify (
-             &gEfiCpuArchProtocolGuid,
-             Event,
-             &Registration
-             );
-  ASSERT_EFI_ERROR (Status);
+  // Status = CoreCreateEvent (
+  //            EVT_NOTIFY_SIGNAL,
+  //            // MU_CHANGE START: Use Project Mu Arch Protocol Notify

```

Comment:
This is commenting out another MU_CHANGE, can we either revert that change if it is no longer applicable in entirety or drop the MU_CHANGE parts of these comments in a separate commit that we can mark as something to fix up on rebase?

Reply:
Reverted this code group to what it is in EDK2 and commented it out.